### PR TITLE
refactor: Migrate ConnectRequestsViewModel to Typescript

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

File: src/script/view_model/content/ConnectRequestsViewModel.ts [hunk 1]
```diff
@@ -17,32 +17,44 @@
...
-window.z.viewModel.content = z.viewModel.content || {};
+type Repositories = {
+  user: UserRepository;
+};
 
-z.viewModel.content.ConnectRequestsViewModel = class ConnectRequestsViewModel {
+export class ConnectRequestsViewModel {
+  mainViewModel: MainViewModel;
+  userRepository: UserRepository;
+  logger: Logger;
```
Comment (bennycode) [inline src/script/view_model/content/ConnectRequestsViewModel.ts]: Logger can be made private.

File: src/script/view_model/content/ConnectRequestsViewModel.ts [hunk 2]
```diff
@@ -17,32 +17,44 @@
...
-window.z.viewModel = z.viewModel || {};
-window.z.viewModel.content = z.viewModel.content || {};
+type Repositories = {
+  user: UserRepository;
+};
 
-z.viewModel.content.ConnectRequestsViewModel = class ConnectRequestsViewModel {
+export class ConnectRequestsViewModel {
+  mainViewModel: MainViewModel;
+  userRepository: UserRepository;
```
Comment (bennycode) [inline src/script/view_model/content/ConnectRequestsViewModel.ts]: Please make everything private which is not used by a Knockout UI component.

File: src/script/view_model/content/ConnectRequestsViewModel.ts [hunk 3]
```diff
@@ -57,7 +69,7 @@ z.viewModel.content.ConnectRequestsViewModel = class ConnectRequestsViewModel {
    * @param {User} request Rendered connection request
    * @returns {undefined} No return value
    */
-  afterRender(elements, request) {
+  afterRender(elements: Object, request: User) {
```
Comment (AndyLnd) [inline src/script/view_model/content/ConnectRequestsViewModel.ts]: Can you please add return types?

File: src/script/view_model/content/ConnectRequestsViewModel.ts [hunk 4]
```diff
@@ -17,32 +17,44 @@
...
+  shouldUpdateScrollbar: ko.Computed<User[]>;
   /**
    * View model for connection requests.
    *
    * @param {MainViewModel} mainViewModel Main view model
    * @param {ContentViewModel} contentViewModel Content view model
    * @param {Object} repositories Object containing all repositories
    */
-  constructor(mainViewModel, contentViewModel, repositories) {
+  constructor(mainViewModel: MainViewModel, contentViewModel: ContentViewModel, repositories: Repositories) {
```
Comment (AndyLnd) [inline src/script/view_model/content/ConnectRequestsViewModel.ts]: You could define them right here as `private readonly mainViewModel: MainViewModel` etc. That would save you the need to declare them (line 36) and also save the this.mainViewModel = mainViewModel.

File: src/script/view_model/content/ConnectRequestsViewModel.ts [hunk 5]
```diff
@@ -17,32 +17,44 @@
...
+import {ContentViewModel} from '../ContentViewModel';
+import {UserRepository} from '../../user/UserRepository';
+import {ActionsViewModel} from '../ActionsViewModel';
+import {User} from '../../entity/User';
 
-window.z = window.z || {};
-window.z.viewModel = z.viewModel || {};
-window.z.viewModel.content = z.viewModel.content || {};
+type Repositories = {
+  user: UserRepository;
```
Comment (AndyLnd) [inline src/script/view_model/content/ConnectRequestsViewModel.ts]: If we just use this one repository, we should pass it directly and not the whole bunch of repositories in this object.

**Pull request merged**